src/js/dom.js

```diff
@@ -0,0 +1,440 @@
+ * @private
+ */
+var delegateEvent = function (selector, event, handler, scope) {
+  (scope || document).addEventListener(event, function(event) {
+    var listeningTarget = closest(event.target, selector);
+    if (listeningTarget) {
+      handler.call(listeningTarget, event);
+    }
+  });
+}
+
+/**
+ * Extends properties of all arguments into a single object.
+ * @param {...Object} objects - Objects to merge
+ * @param {string} event - The event name
+ * @param {function} handler - The event handler function
+ * @returns {Object}
+ * @see {@link https://gomakethings.com/vanilla-javascript-version-of-jquery-extend/}
+ */
+var extend = function (objects) {
+  // Variables
+  var extended = {};
+  var deep = false;
+  var i = 0;
+  var length = arguments.length;
+
+  // Check if a deep merge
+  if ( Object.prototype.toString.call( arguments[0] ) === '[object Boolean]' ) {
+      deep = arguments[0];
+      i++;
+  }
+
+  // Merge the object into the extended object
+  var merge = function (obj) {
+      for ( var prop in obj ) {
+          if ( Object.prototype.hasOwnProperty.call( obj, prop ) ) {
+              // If deep merge and property is an object, merge properties
+              if ( deep && Object.prototype.toString.call(obj[prop]) === '[object Object]' ) {
+                  extended[prop] = extend( true, extended[prop], obj[prop] );
+              } else {
+                  extended[prop] = obj[prop];
+              }
+          }
+      }
+  };
+
+  // Loop through each object and conduct a merge
+  for ( ; i < length; i++ ) {
+      var obj = arguments[i];
+      merge(obj);
+  }
+
+  return extended;
+};
+
+/**
+ * The callback function for getJSON().
+ *
+ * @callback xhrCallback
+ * @param {boolean} success - True on success. False on error.
+ * @param {Object} data - The parsed data. null if success == false
+ * @param {XMLHttpRequest} xhr - The request object
+ * @see getJSON
+ */
+
+/**
+ * Runs an Ajax request against a url and calls the callback function with
+ * the parsed JSON result.
+ * @param {string} url - The url to request
+ * @param {xhrCallback} callback - The callback function
+ * @returns {XMLHttpRequest}
+ */
+var getJSON = function (url, callback) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url, true);
+  xhr.setRequestHeader('Content-Type', 'application/json');
+  xhr.setRequestHeader('Accept', 'application/json');
+
+  xhr.onload = function() {
+    if (xhr.status >= 200 && xhr.status < 400) {
+      var data = JSON.parse(xhr.responseText);
+      callback(true, data, xhr);
+    } else {
+      callback(false, null, xhr);
+    }
+  };
+
+  xhr.onerror = function() {
+    callback(false, null, xhr);
+  };
+
+  xhr.send();
+}
+
+/**
+ * Deserializes JSON values from strings. Used with data attributes.
+ * @param {string} value - String to parse
+ * @returns {Object}
+ * @private
+ */
+var deserializeValue = function (value) {
```

Comment:
What is the reason that these functions are defined globally here? Shouldn't the be part of the `dq` object?
I'm not too familiar with the latest JS module specs, maybe those functions stay local to the file anyway, then forget about my comment. Just wanna be sure.